packages/compass-query-bar/src/constants/query-option-definition.ts

```diff
@@ -33,6 +33,12 @@ export const OPTION_DEFINITION: {
     placeholder: "{ field: -1 } or [['field', -1]]",
     link: 'https://docs.mongodb.com/manual/reference/method/cursor.sort/',
   },
+  hint: {
+    name: 'hint',
```

Comment:
I'd consider using `Index Hint` or similar for the UI, somebody who isn't familiar with this database feature might not be aware of what kind of `Hint` this is 🙂

Reply:
This makes perfect sense, added it! I noticed though that the `label` field was not being used in https://github.com/mongodb-js/compass/pull/5666/commits/0eae3edfbcc920bb355e24922b9e6a3344054bbd#diff-2d6d3eaf924ff6725b2a090da228b745956458a31e338881ae9d29ab5d712f6bR178 so I added that as well. I believe this also changes how `Max Time MS` is displayed (before it was MaxTimeMS) since it's using a label property as well

![image](https://github.com/mongodb-js/compass/assets/6215162/b8a8a2f4-e34c-4436-8b36-ce07dd8b46f3)